CONTRIBUTING.md

```diff
@@ -1,106 +1,186 @@
+| Client Libraries     | [![Build Status](https://dev.azure.com/azure-sdk/public/_apis/build/status/41?branchName=master)](https://dev.azure.com/azure-sdk/public/_build/latest?definitionId=41&branchName=master) |
 
-### Prerequisites:
-  Install VS 2017 (Professional or higher) and make sure you have the latest updates (https://www.visualstudio.com/).
+# Prerequisites:
 
-### To build:
-=======
+Install VS 2019 (Professional or higher) and make sure you have the latest updates (https://www.visualstudio.com/).
 
-#### If you are building from VS, add a nuget feed source that points to < root >\tools\LocalNugetFeed directory
- 1. Open any solution, eg "Compute\Microsoft.Azure.Management.Compute.sln"
- 2. Build solution from VS
- 
-#### Full Build from command line
+**Client Libraries** are sdks used to interact with azure resources at application run time while **Management Libraries** are those used to manage (create/modify/delete) Azure resources.
+<br/><br/>
 
- 1. Open VS 2017 developer command prompt
- 2. Navigate to repository root directory
- 3. Invoke **msbuild** build.proj /t:Build
- will Build
- ##### *Build* without any scope will build all management SDK's.
+# Client Libraries
 
-#### Build single SDK
-`msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
+## TO BUILD:
+
+### Single Service from Command Line
+
+1. Open VS 2019 Command Prompt
+2. Navigate to service directory e.g. _"sdk\eventhub"_
+3. Invoke `dotnet build`
+4. or Build the **service.proj** in the repo root, passing the directory name of the specific service as a property. e.g. `dotnet build eng\service.proj /p:ServiceDirectory=eventhub`
+
+### Single Service from Visual Studio
+
+1. Open any data-plane solution e.g. _"sdk\eventhub\Microsoft.Azure.EventHubs.sln"_
+2. Build solution from Visual Studio
+
+### All Client Services from Command Line
+
+1. Open VS 2019 Command Prompt
+2. Navigate to repository root directory
+3. Invoke `dotnet build eng\service.proj`
+
+## TO TEST:
+
+### Single Service from Command Line
+
+1. Open VS 2019 Command Prompt
+2. Navigate to service directory e.g. _"sdk\eventhub"_
+3. Invoke `dotnet test --filter TestCategory!=Live` _(Skips live tests)_
+4. or run test against **service.proj** in the repo root, passing the directory name of the specific service as a property. e.g. `dotnet test eng\service.proj /p:ServiceDirectory=eventhub --filter TestCategory!=Live`
+
+### Single Service from Visual Studio
+
+1. Build.
+2. Test Explorer window will get populated with tests. Select test and `Run` or `Debug`
+
+### All Client Services from Command Line
+
+1. Open VS 2019 Command Propmpt
+2. Navigate to repository root directory
+3. Invoke `dotnet test eng\service.proj --filter TestCategory!=Live`
+   <br/><br/>
+
+# Management Libraries
+## TO BUILD:
+
+### If you are building from Visual Studio, add a nuget feed source that points to < root >\tools\LocalNugetFeed directory
```

Comment:
@shahabhijeet do people still need to LocalNugetFeed? If they do we should get the tools published and eliminate this need.

Reply:
Yes we do need localnugetfeed for some of the nuget packages that are being used on all management plane test projects.
The transition team knows about to move to a different version, but until that happens, localnugetfeed should be there.
Plus long term, this directory allows us to take advantage of do local testing or add a nuget package that will be used by the entire repo (e.g. tools pacakge)

Reply:
As we chatted I think keeping the directory is fine until the older dependencies can be removed but I would still advise against keeping it long term because it sets up a bad example. If people want to do some temporary local testing they can easily create a directory and add it to their config. Committing packages into the repo should generally be avoided.